R/test-DataClass.R

```diff
@@ -0,0 +1,188 @@
+#' column is a character in the cleaned data (data$clean)
+#' @param data The clean data to check
+#' @param level character_array the level of the data to check
+expect_clean_cols <- function(data, level) {
+  testthat::expect_s3_class(data[["date"]], "Date")
+  level_region_str <- paste0("level_", level, "_region")
+  testthat::expect_type(data[[level_region_str]], "character")
+}
+
+#' Test that processed columns contain the correct data and types
+#' @description Checks that processed data columns date, cases_new, cases_total,
+#' deaths_new, deaths_total and that region level have the correct types.
+#' @param data The data to check
+#' @param level character_array the level of the data to check
+#' @param localised logical to check localised data or not, defaults to
+#' TRUE.
+expect_processed_cols <- function(data, level = "1", localised = TRUE) {
+  testthat::expect_s3_class(data[["date"]], "Date")
+  testthat::expect_type(data[["cases_new"]], "double")
+  testthat::expect_type(data[["cases_total"]], "double")
+  testthat::expect_type(data[["deaths_new"]], "double")
+  testthat::expect_type(data[["deaths_total"]], "double")
+  if (!localised) {
+    level_region_str <- paste0("level_", level, "_region")
+    testthat::expect_type(data[[level_region_str]], "character")
+  }
+}
+
+#' Test that cleaned columns contain data/
+#' @description Checks that cleaned columns cases, deaths, recovered and test
+#' (new and total) are not entirely composed of NAs.
+#' @param cntry_obj The DataClass object (R6Class) object to perform checks on.
+#' @importFrom purrr map walk
+#' @importFrom dplyr filter
+#' @importFrom rlang !!
+expect_columns_contain_data <- function(cntry_obj) {
+  cols_present <- function(col) {
+    if (length(cntry_obj$source_data_cols[grep(
+      col, tolower(cntry_obj$source_data_cols)
+    )]) > 0) {
+      return(paste0(col, c("_new", "_total")))
+    } else {
+      return(NULL)
+    }
+  }
+  cols <- c("cases", "deaths", "recovered", "test")
+  cols2check <- map(cols, cols_present)
+  cols2check <- unlist(cols2check)
+  walk(
+    cols2check,
+    ~ {
+      testthat::test_that(
+        paste0(
+          cntry_obj$data_name, "column '", .x, "' is not just composed of NA"
+        ),
+        {
+          testthat::expect_true(
+            nrow(cntry_obj$data$processed %>% filter(!is.na(!!.x))) > 0
+          )
+        }
+      )
+    }
+  )
+}
+
+#' Test download method works correctly
+#' @description Test data can be downloaded if download = TRUE, or a requested
+#' snapshot file is not found, and store a snap shot at the snapshot_path. If an
+#' existing snapshot file is found then load this data to use in future tests.
+#' @param cntry_obj The DataClass object (R6Class) object to perform checks on.
+#' @param download logical check to download or use a snapshot of the data
+#' @param snapshot_path character_array the path to save the downloaded
+#' snapshot to.
```

Comment:
add to export and make a new docs family called test + add to pgkdown site

Reply:
Iv'e added a family tag of tests and an export tag on the `test-DataClass` functions, how do I add this to the pkgdown site?